pkg/controllers/workspace_controller.go

```diff
@@ -245,10 +293,19 @@ func (c *WorkspaceReconciler) applyWorkspaceResource(ctx context.Context, wObj *
 		}
 	}
 
-	if err = c.updateStatusConditionIfNotMatch(ctx, wObj, kaitov1alpha1.WorkspaceConditionTypeMachineStatus, metav1.ConditionTrue,
-		"installNodePluginsSuccess", "machines plugins have been installed successfully"); err != nil {
-		klog.ErrorS(err, "failed to update workspace status", "workspace", klog.KObj(wObj))
-		return err
+	if featuregates.FeatureGates[consts.FeatureFlagKarpenter] &&
```

Comment:
Same here, if featuregate is enabled, the code should go to else path.

Reply:
The feature flag condition should go to the else (addressed). Otherwise, the logic is correct.